src/librustc_mir/transform/mod.rs

```diff
@@ -493,9 +499,7 @@ fn run_optimization_passes<'tcx>(
         ],
     );
 
-    // All other optimizations can assume there will be no generator aggregates.
-    // At the end some cleanup is performed to remove constructs that are uninteresting for
-    // codegen backends.
```

Comment:
The last two lines of this comment, are they still accurate? If yes they should be added at the right spot in the `optimizations` list.

Reply:
There's a `Some cleanup necessary at least for LLVM and potentially other codegen backends.` comment on the codgen cleanup optimization list

Reply:
But that was never "at the end", was it?
Probably this comment is from before we had actual MIR optimizations, and should be removed.

Reply:
It is at the end. We need to run the critical edges pass, otherwise llvm doesn't know where to put its stuff after a function call.

Reply:
I cannot even find this diff in the main diff view any more?
I thought this comment was next to a `run_pass`, where should not be; it should be next to the corresponding pass list definition.

Reply:
Oh I guess this is a comment that you added and then removed again. I saw it as removed in the diff and thought it existed before the PR. Of course GH fails completely when comments are attached to commits in a PR, there's now ay to figure out *which* commit after the fact.

Never mind then.